### PR TITLE
Add the api_ao_messages#index action

### PR DESCRIPTION
**Pull request by ysbaddaden**
Exposes the `ao_messages#get_ao` to the OAuth 2.0 API.
Requires the `token` attribute to be defined.

### CODE REVIEW

File: app/controllers/api/api_ao_messages_controller.rb [hunk 1]
```diff
@@ -2,6 +2,12 @@ class ApiAoMessagesController < ApiAuthenticatedController
   before_filter :require_account_and_application!
   include AoMessageCreateCommon
 
+  rescue_from(KeyError) { head :bad_request }
+
+  def index
+    render :json => AoMessage.find_all_by_application_id_and_token(@application.id, params.fetch(:token))
```
Comment (matiasgarciaisaia) [inline app/controllers/api/api_ao_messages_controller.rb:8]: Would this include pagination?

We usually don't need it, but we've also seen some tokens in prod shared between +2k AOs.

Comment (ysbaddaden) [reply app/controllers/api/api_ao_messages_controller.rb:8]: I missed your comment @matiasgarciaisaia: there is no pagination for the time being, I didn't think about that.

**Comment by ysbaddaden** [comment]
@matiasgarciaisaia Now with pagination of the AO messages. Since we didn't have pagination in the API (yet) I went to my goto solution: the Link HTTP header, following a bunch of APIs (e.g. GitHub):

```
Link: <http://nuntium.instedd.org/api/ao_messages?token=<redacted>&page=2>; rel="next"
```

I also set the pagination to 50 messages so we shouldn't have to worry about it in Surveda.

**Review by matiasgarciaisaia — APPROVED** [review]
Looks good!

🚀 :shipit:

**Pull request merged**